react/features/security/components/security-dialog/web/PasswordForm.tsx

```diff
@@ -43,163 +42,74 @@ interface IProps extends WithTranslation {
-     */
-    enteredPassword: string;
-}
-
 /**
  * React {@code Component} for displaying and editing the conference password.
  *
- * @augments Component
+ * @returns {ReactElement}
  */
-class PasswordForm extends Component<IProps, IState> {
-    /**
-     * Implements React's {@link Component#getDerivedStateFromProps()}.
-     *
-     * @inheritdoc
-     */
-    static getDerivedStateFromProps(props: IProps, state: IState) {
-        return {
-            enteredPassword: props.editEnabled ? state.enteredPassword : ''
-        };
-    }
-
-    state = {
-        enteredPassword: ''
-    };
-
-    /**
-     * Initializes a new {@code PasswordForm} instance.
-     *
-     * @param {IProps} props - The React {@code Component} props to initialize
-     * the new {@code PasswordForm} instance with.
-     */
-    constructor(props: IProps) {
-        super(props);
-
-        // Bind event handlers so they are only bound once per instance.
-        this._onEnteredPasswordChange = this._onEnteredPasswordChange.bind(this);
-        this._onKeyPress = this._onKeyPress.bind(this);
-    }
-
-    /**
-     * Implements React's {@link Component#render()}.
-     *
-     * @inheritdoc
-     * @returns {ReactElement}
-     */
-    render() {
-        return (
-            <div className = 'info-password'>
-                {this._renderPassword()}
-                {this._renderPasswordField()}
-            </div>
-        );
-    }
+export default function PasswordForm({
+    editEnabled,
+    locked,
+    onSubmit,
+    password,
+    passwordNumberOfDigits,
+    visible
+}: IProps) {
+    const { t } = useTranslation();
+    const [ enteredPassword, setEnteredPassword ] = useState('');
+    const onKeyPress = useCallback(event => {
+        if (event.key === 'Enter') {
+            event.preventDefault();
+            event.stopPropagation();
 
-    /** .........
-     * Renders the password if there is any.
-     *
-     * @returns {ReactElement}
-     */
-    _renderPassword() {
-        const { locked, t } = this.props;
+            onSubmit(enteredPassword);
+        }
+    }, [ onSubmit, enteredPassword ]);
 
-        return locked && <>
-            <span className = 'info-label'>
-                {t('info.password')}
-            </span>
-            <span className = 'spacer'>&nbsp;</span>
-            <span className = 'info-password-field info-value'>
-                {locked === LOCKED_LOCALLY ? (
-                    <div className = 'info-password-local'>
-                        {this.props.visible ? this.props.password : '******' }
-                    </div>
-                ) : (
-                    <div className = 'info-password-remote'>
-                        {this.props.t('passwordSetRemotely')}
-                    </div>
-                ) }
-                {this._renderPasswordField()}
-            </span>
-        </>;
+    if (!editEnabled && enteredPassword && enteredPassword !== '') {
+        setEnteredPassword('');
```

Comment:
Why this?

Reply:
this is the same as the getDerivedPropsToState method before. It will basically reset the password in the state if we go in the non-edit mode.